pkg/daemon/daemon.go

```diff
@@ -328,6 +331,12 @@ func (dn *Daemon) Run(stopCh <-chan struct{}, exitCh <-chan error) error {
 			if err := dn.tryCreateUdevRuleWrapper(); err != nil {
 				glog.V(2).Info("Could not create udev rule: ", err)
 			}
+			// Mellanox plugin is loaded when mellanox NICs exist
+			if _, ok := dn.LoadedPlugins[mellanoxPlugin]; ok {
```

Comment:
so I don't like this approach.
can you check if we can run go routine in the mellanox plugin?

Reply:
In mellanox plugin it is possible to check/set VF admin  and node_guid which will ubind/bind the VF driver. this lead the VF will move back to host namepace.
In that case the sriov pods will not be functionining and we need to drain the node and restart device plugin which is not feasible by mellanox plugin

Reply:
I am talking about running monitor thread from the plugin shared object. Is it possible?

Reply:
@moshe010 It is possible to run the monitor thread from the plugin, but it is not possible to expose a new function from the plugin object

Reply:
Can you start it on  OnNodeStateChange is it has RDMA=true and strop it if there is not?

https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/master/pkg/plugins/mellanox/mellanox_plugin.go#L71